### PR TITLE
use fromValue in more places in builtins

### PR DESCRIPTION
**Pull request by hSloan**
Do this to avoid pattern matching and throwing errors explicitly. 

Thanks for allowing me to contrib ^_^

### CODE REVIEW

File: src/Nix/Builtins.hs [hunk 1]
```diff
@@ -664,8 +649,8 @@ getEnv_ = fromNix >=> \s -> do
         Just v  -> Text.pack v
 
 sort_ :: MonadBuiltins e m => NThunk m -> NThunk m -> m (NValue m)
-sort_ comparator list = force list $ \case
-    NVList l -> NVList <$> sortByM cmp l
+sort_ comparator list = force list $ fromValue >=> \l ->
+    NVList <$> sortByM cmp l
```
Comment (jwiegley) [inline src/Nix/Builtins.hs:653]: you can use `toNix =<<` instead of `NVList <$>` in places like this.

**Comment by jwiegley** [comment]
Thank you!

How is `genericClosure` treating you? :)

**Comment by jwiegley** [comment]
@hSloan Oh, please hold off on further `Builtins.hs` changes like this. I'm working on a rather large refactoring within that file right now. You'll see the push when it comes through.

**Comment by jwiegley** [comment]
I'm all done.

**Comment by hSloan** [comment]
@jwiegley Will do, I'm planning on finishing off the `genericClosure` ticket next.

**Pull request merged**